text/0107-construct-library-module-lifecycle.md

```diff
@@ -0,0 +1,146 @@
+    * **Repeat** - These activities are repeated until the module owner decides the module is feature complete and the API is ergonomic. At the end of the experimental stage, the construct library module’s GitHub tracking issue will have a description of the service, a list of targeted use cases, and a list of L2 constructs required to satisfy the use cases.
+* **Exit Criteria**:
+    * **Feature complete** - The module owner declares the module feature complete for targeted use cases, and the API is ergonomic. 
+        >**Note**: Feature complete does not mean 100% AWS service feature coverage, instead, it means the module owner thinks we’ve implemented the minimum features required for a delightful developer experience for modeling the service infrastructure as code.
+     * **Successful API review** - conduct an API review/developer experience demo and invite relevant stakeholders
+        * use module's README as the API review guide
+        * address any concerns about API ergonomics
+        * review all AWS Lint exceptions to ensure they are acceptable for dev preview
+        * ensure at least 90% unit test coverage for all implemented L2 constructs
+        * if the API/DevEx is rejected, make a list of necessary remediation tasks and update the module's tracking issue 
+
+## Stage 2 - Developer Preview 
+
+The goal of developer preview is to deliver a release candidate with a stable API to conduct user acceptance testing and accumulate sufficient usage to declare a module is ready for general availability. We will only make breaking changes to developer preview modules when required to address unforeseen customer use cases or issues. Since the potential for breaking changes still remains while in Developer Preview, the package name will keep the “-experimental” suffix during this stage.
+
+* **Stage 2 activities**:
+    * **Update Roadmap** - move to the CDK public roadmap “Developer Preview” column to indicate we need customer feedback on the module’s implementation
+    * **Observe** - monitor module utilization and GitHub issues filed against the experimental module
+    * **Quality Assurance** - fix all P0 bugs as they are  discovered and review all P1 bugs, making a clearly communicated fix/defer decision for each one 
+* **Exit Criteria**:
+    * **Bake time** - modules will spend at least 3 months in developer preview to accumulate sufficient usage data to gain confidence in the API’s ergonomics and customer acceptance
+    * **Usage** - module utilization in 2,000 or more stacks is considered sufficient usage. We will evaluate the language distribution of module use to determine if we need to target specific underrepresented programming language communities for feedback.
+    * **No P0 bugs** - no new P0 bugs in the past 4 weeks
+    * **Successful API/Docs review** - conduct the final API review/developer experience demo and invite relevant stakeholders
```

Comment:
If we add the `ga` label i mentioned before, this would be the point we go over all issues labeled with `ga` to make sure we either implemented them or are knowingly dropping them from the release (removing the label)

Reply:
Yep, makes sense. Using the `ga` label or all P1s, either way this is the time to make the final decision on the functionality to ship.